src/components/TreeSelect/TreeSelect.tsx

```diff
@@ -59,28 +55,104 @@ export const TreeSelect: React.FC<Props> = ({
         .map((value) => findNode({ value, tData }).target.title)
         .join(", ");
 
+  useEffect(() => {
+    setOptionsLabel(optionsLabel);
+  }, [optionsLabel, setOptionsLabel]);
+
+  if (!isVisible) {
+    return null;
+  }
+
   return (
-    <Wrapper data-cy={dataCy} width={width}>
-      <BarWrapper onClick={toggleOptions} className="cy-treeselect-bar">
-        <LabelWrapper>
-          {inputLabel}
-          {optionsLabel || "No filters selected"}
-        </LabelWrapper>
-        <ArrowWrapper>
-          <div>
-            <Icon glyph={isVisible ? "ChevronUp" : "ChevronDown"} />
-          </div>
-        </ArrowWrapper>
-      </BarWrapper>
-      {isVisible && (
+    <ConditionalWrapper
+      condition={isDropdown}
+      wrapper={(children) => (
         <RelativeWrapper>
           <OptionsWrapper data-cy="tree-select-options">
-            {renderCheckboxes({ state: filteredState, tData, onChange })}
+            {children}
           </OptionsWrapper>
         </RelativeWrapper>
       )}
-    </Wrapper>
+    >
+      <>
+        {renderCheckboxes({
+          state: filteredState,
+          tData,
+          onChange,
+        })}
+      </>
+    </ConditionalWrapper>
+  );
+};
+
+// depth first traversal checkbox data.
+// pushes parent then children to rows array
+// keeps track of level for indentation
+const renderCheckboxes = ({
+  tData,
+  state,
+  onChange,
+}: {
+  tData: TreeDataEntry[];
+  state: string[];
+  onChange: (v: [string]) => void;
+}): JSX.Element[] => {
+  const rows: JSX.Element[] = [];
+  tData.forEach((entry) => {
+    renderCheckboxesHelper({ rows, data: entry, onChange, state, tData });
+  });
+  return rows;
+};
+
+const renderCheckboxesHelper = ({
+  rows,
+  data,
+  onChange,
+  state,
+  tData,
+}: {
+  rows: JSX.Element[];
+  data: TreeDataEntry;
+  onChange: (v: string[]) => void;
+  state: string[];
+  tData: TreeDataEntry[];
+}): void => {
+  const ParentCheckboxWrapper = getCheckboxWrapper(0);
+  // push parent
+  const onChangeFn = (): void =>
+    handleOnChange({ state, value: data.value, onChange, tData });
+  rows.push(
+    <ParentCheckboxWrapper key={data.key}>
+      <Checkbox
+        className="cy-checkbox"
+        onChange={onChangeFn}
+        label={data.title}
+        checked={state.includes(data.value)}
+        bold={false}
+        data-cy="checkbox"
+      />
+    </ParentCheckboxWrapper>
   );
+  // then examine children
+  const ChildCheckboxWrapper = getCheckboxWrapper(1);
+  if (data.children) {
+    data.children.forEach((child) => {
+      const onChangeChildFn = (): void =>
+        handleOnChange({ state, value: child.value, onChange, tData });
+      rows.push(
+        <ChildCheckboxWrapper key={child.key}>
```

Comment:
i think this line results in a dupe key error for the aborted status. you can see it in the console when opening the dropdown on http://localhost:3000/commits/spruce while pointing to prod